exotic/exotic.py

```diff
@@ -1334,12 +1334,8 @@ def process_dark_frames(dark_files):
     # Dark files whose median is much higher than the overall dark files median will be filtered
     # e.g. to discard saturated dark files that may negatively affect the master dark used to calibrate the science frames
     # First pass: collect all dark frame medians
-    darks_medians = []
-    for dark_file in dark_files:
-        dark_data = fits.getdata(dark_file)
-        dark_median = np.nanmedian(dark_data)
-        darks_medians.append((dark_file, dark_median))
-        
+    darks_medians = [(dark_file, np.nanmedian(fits.getdata(dark_file))) for dark_file in dark_files]
+
```

Comment:
Might be an error here. `dark_file` should really be `fits.getdata(dark_file)`, as you are doing when you get the median here. I didn't mind not making this a list comprehension in the previous review I gave as it may have made it more unreadable. I'll let you decide which way is best (current or old). However, I do want the possible bug fixed.

Reply:
I mean the first `dark_file` in `(dark_file, np.nanmedian(fits.getdata(dark_file)))`